File: admin-web/src/types/collection.ts
export interface RestaurantCollection {
  id: string;
  title: string;
  subtitle: string | null;
  sortOrder: number;
  isPublished: boolean;
  createdAt: Date;
}

export interface CollectionItem {
  id: string;
  collectionId: string;
  restaurantId: string;
  restaurantName: string | null;
  note: string | null;
  sortOrder: number;
}

export function parseRestaurantCollection(
  raw: Record<string, unknown>,
): RestaurantCollection {
  return {
    id: String(raw.id),
    title: String(raw.title ?? ""),
    subtitle: raw.subtitle != null ? String(raw.subtitle) : null,
    sortOrder: Number(raw.sort_order ?? 0) || 0,
    isPublished: Boolean(raw.is_published),
    createdAt: new Date(String(raw.created_at)),
  };
}

export function parseCollectionItem(raw: Record<string, unknown>): CollectionItem {
  return {
    id: String(raw.id),
    collectionId: String(raw.collection_id),
    restaurantId: String(raw.restaurant_id),
    restaurantName: raw.restaurant_name != null ? String(raw.restaurant_name) : null,
    note: raw.note != null ? String(raw.note) : null,
    sortOrder: Number(raw.sort_order ?? 0) || 0,
  };
}

export interface CollectionCommentAdmin {
  id: string;
  collectionId: string;
  collectionTitle: string;
  content: string;
  nickname: string;
  isHidden: boolean;
  createdAt: Date;
}

export function parseCollectionCommentAdmin(
  raw: Record<string, unknown>,
): CollectionCommentAdmin {
  return {
    id: String(raw.id),
    collectionId: String(raw.collection_id),
    collectionTitle: String(raw.collection_title ?? ""),
    content: String(raw.content ?? ""),
    nickname: String(raw.nickname ?? "탈퇴한 사용자"),
    isHidden: Boolean(raw.is_hidden),
    createdAt: new Date(String(raw.created_at)),
  };
}

File: lib/models/collection.dart
class RestaurantCollection {
  final String id;
  final String title;
  final String? subtitle;
  final int sortOrder;
  final int likeCount;
  final bool likedByMe;
  final int commentCount;

  const RestaurantCollection({
    required this.id,
    required this.title,
    this.subtitle,
    required this.sortOrder,
    this.likeCount = 0,
    this.likedByMe = false,
    this.commentCount = 0,
  });

  factory RestaurantCollection.fromMap(Map<String, dynamic> map) {
    return RestaurantCollection(
      id: map['id'] as String,
      title: map['title'] as String,
      subtitle: map['subtitle'] as String?,
      sortOrder: map['sort_order'] as int? ?? 0,
      likeCount: map['like_count'] as int? ?? 0,
      likedByMe: map['liked_by_me'] as bool? ?? false,
      commentCount: map['comment_count'] as int? ?? 0,
    );
  }

  RestaurantCollection copyWith({int? likeCount, bool? likedByMe, int? commentCount}) {
    return RestaurantCollection(
      id: id,
      title: title,
      subtitle: subtitle,
      sortOrder: sortOrder,
      likeCount: likeCount ?? this.likeCount,
      likedByMe: likedByMe ?? this.likedByMe,
      commentCount: commentCount ?? this.commentCount,
    );
  }
}

class CollectionItem {
  final String id;
  final String restaurantId;
  final String? note;
  final int sortOrder;

  const CollectionItem({
    required this.id,
    required this.restaurantId,
    this.note,
    required this.sortOrder,
  });

  factory CollectionItem.fromMap(Map<String, dynamic> map) {
    return CollectionItem(
      id: map['id'] as String,
      restaurantId: map['restaurant_id'] as String,
      note: map['note'] as String?,
      sortOrder: map['sort_order'] as int? ?? 0,
    );
  }
}
